### DISCUSSION TITLE
Export Grain Diametres

### DISCUSSION THREAD

**Discussion by janzenchoi** [neperfepx/neper#524] [Q&A] [answered]
Hello,

I generated this 2D tessellation earlier of size 300x300. I have highlighted the grain corresponding to id 8.

![image](https://user-images.githubusercontent.com/75835751/180123126-641fed08-259c-4861-a3ec-a050e0df6b2e.png)

When I try to export the information about each statistic via `-statcell diameq,sphericity`, the values seem off. I have attached the exported information below.

`1 223.799471179171 0.907622468533`
`2 180.157538502079 0.882413954969`
`3 111.135850788852 0.869609498953`
`4 97.385111537634 0.877180104814`
`5 78.937009584651 0.829346277410`
`6 48.382797552197 0.904909962505`
`7 7.865692789547 0.776350433558`
`8 8.230832485491 0.777066088058`
`9 14.654291042968 0.885385820477`
`10 17.694692005741 0.876241164540`
`11 31.371437462168 0.927315205802`

The exported file shows that the diametre of the grain is around `8.23`. Comparing the size of the grain with the length of the tessellation (`300`), I feel like the exported diametres are off. Do you know why this might be the case?

**Comment by rquey** [comment, marked as answer]
Please (always) provide the Neper command (and other inputs) and terminal output, so that what you are showing can be reproduced.

**Reply by janzenchoi** [reply]
Apologies. These are the following steps I used.

1) Tessellated the parent grains:
`neper -T -n from_morpho -morpho "diameq:lognormal(19.24038,19.577382869014947)+lognormal(123.760806,70.17013501483378),1-sphericity:lognormal(0.055885,0.09567653839891993)" -morphooptiini coo:packing,weight:radeq -dim 2 -domain "square(300,300)" -statcell diameq,sphericity -o parent.tess`

2) Visualised the tessellation of the parent grains:
`neper -V parent.tess -print parent`

3) Generated a file of lamellae widths, named `twin_width`:
`1 300`
`2 300`
`3 300`
`4 300`
`5 300`
`6 300`
`7 300`
`8 10`
`9 300`
`10 300`
`11 300`

4) Incorporated the lamellae widths into the tessellation of the parent grains:
`neper -T -n 11::from_morpho -morpho "voronoi::lamellar(w=file(twin_width),v=crysdir(1,0,0))" -morphooptiini "file(parent.tess)" -format tess,tesr -tesrsize 150 -tesrformat ascii -dim 2 -domain "square(300,300)" -o twinned.tess`

5) Visualised the tessellation of the parent grains:
`neper -V twinned.tess -print twinned`

The above steps should produce a `300x300` tessellation, with the 8th grain (`id = 8`) having lamellae of `width = 10`. However, in the `.stcell` file, it shows the equivalent diametre of the 8th grain to be around `8`. I have attached the visualisations below.

![image](https://user-images.githubusercontent.com/75835751/180160187-437fef20-62d7-4f81-b59c-909466566b92.png)

I can't seem to wrap my head around why this happens. Another weird thing is that if the first width (i.e., width value corresponding to the first grain) is not a large number (e.g., 300), then Neper will not be able to incorporate the lamellae into the tessellation of the parent grain (i.e., `step 4)` will fail).

Thank you once again for the assistance.

**Reply by rquey** [reply]
@jazzzmannn Next step for a good bug report is to provide a *minimal working example*: can some of the options be removed without affecting the behavior?  Can the number of scales be reduced, or the number of grains? are the type of questions to consider. It is necessary to isolate the problem, and it is better if who asks does it (of course).

**Reply by janzenchoi** [reply]
Thanks for the advice. I think `-morphooptiini coo:packing,weight:radeq` might be the cause of the first problem (i.e., exported diametres not being actual values).

**Reply by rquey** [reply]
Okay. I would be surprised if there was a problem with the grain diameter data. If you still see a problem, please file an [issue](https://github.com/neperfepx/neper/issues/new/choose) and provide a *minimal working example*.
